ec2system/ec2machine.go

```diff
@@ -1042,6 +1045,41 @@ func (*System) KeepaliveConfig() (period, timeout, rpcTimeout time.Duration) {
 	return
 }
 
+// monitorSpotActions monitors spot instance actions and logs them. In
+// particular, this logs spot instance terminations to help users differentiate
+// spot instance terminations from other termination conditions (e.g. OOM
+// errors, panics, etc.).
+func monitorSpotActions(ctx context.Context) {
+	// We should get a spot instance termination notice two minutes before
+	// termination[0], so polling every thirty seconds should guarantee that we
+	// log it.
+	// [0] https://aws.amazon.com/blogs/aws/new-ec2-spot-instance-termination-notices/
+	tick := time.NewTicker(30 * time.Second)
+	for {
+		select {
+		case <-tick.C:
+		case <-ctx.Done():
+			return
+		}
+		resp, err := http.Get("http://169.254.169.254/latest/meta-data/spot/instance-action")
+		if err != nil {
+			continue
+		}
+		func() {
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+				return
+		}
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Error.Printf("error reading meta-data/spot/instance-action response body: %v", err)
+				return
+		}
+		log.Printf("spot instance action: %v", string(b))
```

Comment:
indentation

Reply:
Oops; my formatter bit the dust without my noticing.